actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java

```diff
@@ -641,6 +639,22 @@ private void processUserRequest(
     TelemetryUtil.telemetryProcessingCall(userMap, targetObject, correlatedObject);
   }
 
+  private void setStateValidation(Map<String, Object> userMap) {
+    String rootOrgId = (String) userMap.get(JsonKey.ROOT_ORG_ID);
+    String custodianRootOrgId = getCustodianRootOrgId();
+    if(!rootOrgId.equals(custodianRootOrgId)) {
+      userMap.put("isStateValidated", true);
+    } else {
+      userMap.put("isStateValidated", false);
+    }
+  }
+
+  private String getCustodianRootOrgId() {
+    String custodianChannel =
```

Comment:
We can do cache of custodianRootOrgId , in a variable.
Every time before making this call will first check value from that variable , if not found then only make this call.

Reply:
yes, checking in userService.getCustodianChannel